### PR TITLE
Added check to prevent users with same name to enter same chatroom

### PR DESCRIPTION
**Pull request by Vivek-blip**

### CODE REVIEW

File: cnew.py [hunk 1]
```diff
@@ -52,15 +52,24 @@ async def producer_handler(cphrsuit, websocket, username, chatroom, servaddr):
...
 
 async def hello(servaddr, username, chatroom, password):
     async with websockets.connect(servaddr) as websocket:
-        cphrsuit = fernetst(password.encode("utf8"))
-        prod = asyncio.get_event_loop().create_task(producer_handler(cphrsuit, websocket, str(username), str(chatroom), str(servaddr)))
-        cons = asyncio.get_event_loop().create_task(consumer_handler(cphrsuit, websocket, str(username), str(chatroom), str(servaddr)))
-        await websocket.send(username+sepr+chatroom)
-        await prod
-        await cons
+        chkUserPresense = await chk_username_presence(websocket,username, chatroom)
```
Comment (vinmay) [inline cnew.py:62]: There is typo. It needs to be 'chkUserPresence'.

File: cnew.py [hunk 2]
```diff
@@ -52,15 +52,24 @@ async def producer_handler(cphrsuit, websocket, username, chatroom, servaddr):
         senddata = cphrsuit.encrjson(senddata)
         await websocket.send(senddata)
 
+async def chk_username_presence(web_socket, user_name, chat_room):
+    await web_socket.send("CHKUSR"+sepr+user_name+sepr+chat_room)
+    async for recvdata in web_socket:
+        return recvdata
```
Comment (t0xic0der) [inline cnew.py:58]: Nice way of adopting the separators in this operation. :+1:

File: cnew.py (hunk 1 above)
Comment (Vivek-blip) [reply cnew.py:62]: Oh, thanks for reporting. I'll correct it

File: main.py [hunk 3]
```diff
@@ -37,14 +37,24 @@ async def notify_mesej(message):
...
 async def chatroom(websocket, path):
     if not chekusav(websocket):
         USERS[websocket] = ""
     try:
         async for mesgjson in websocket:
             if sepr in mesgjson and websocket in USERS:
-                if USERS[websocket] == "":
+                if (mesgjson.split(sepr)[0] == "CHKUSR") & (len(mesgjson.split(sepr)) == 3) :
+                    await websocket.send(str(chk_username_presence(mesgjson)))
+                elif USERS[websocket] == "":
```
Comment (t0xic0der) [inline main.py:57]: Correctly conveyed the user that a user with the same name already exists. :+1:

**Comment by Vivek-blip** [comment]
I've done the correction

**Review by t0xic0der — APPROVED** [review]
The changes you made work the way they should. Thank you.

File: cnew.py [hunk 4]
```diff
@@ -59,7 +59,7 @@ async def chk_username_presence(web_socket, user_name, chat_room):
 
 async def hello(servaddr, username, chatroom, password):
     async with websockets.connect(servaddr) as websocket:
-        chkUserPresense = await chk_username_presence(websocket,username, chatroom)
+        chkUserPresence = await chk_username_presence(websocket,username, chatroom)
```
Comment (t0xic0der) [inline cnew.py:62]: The server-end reports tons of errors on the occassion of user joining/leaving and chat transfers.

image

Take a look.

**Review by t0xic0der — CHANGES_REQUESTED** [review]
Unfortunately, your code brings regressions to the server. Please check the code and make the necessary corrections.

File: cnew.py (hunk 1 above)
Comment (t0xic0der) [reply cnew.py]: Thanks, @vinmay for reporting the typo.

File: cnew.py (hunk 4 above)
Comment (Vivek-blip) [reply cnew.py:62]: I'll be back with the corrections.

Comment (Vivek-blip) [reply cnew.py:62]: I have resolved the above issue

**Comment by Vivek-blip** [comment]
> The changes you made work the way they should. Thank you.

Looking forward to merging if there's no issues 😊

File: main.py [hunk 5]
```diff
@@ -48,14 +48,30 @@ def wrap_text(message, max_width, indent=24):
...
         async for mesgjson in websocket:
             if sepr in mesgjson and websocket in USERS:
-                if USERS[websocket] == "":
+                if (mesgjson.split(sepr)[0] == "CHKUSR") & (len(mesgjson.split(sepr)) == 3) :
+                    result = str(chk_username_presence(mesgjson))
+                    await websocket.send(result)
+                    if(result == "True"):
+                        await websocket.close()
+                        USERS.pop(websocket)
+                        return
```
Comment (t0xic0der) [inline main.py:72]: Does this return serve any practical function?

**Review by t0xic0der — CHANGES_REQUESTED** [review]

**Review by t0xic0der — APPROVED** [review]
I double-checked the parts. Nice work done here.

File: main.py (hunk 5 above)
Comment (Vivek-blip) [reply main.py:72]: Returning will remover the listening instance for that specific user's websocket who couldn't join if I'm not wrong. Please correct me if I'm wrong 😅

**Comment by Vivek-blip** [comment]
> I double-checked the parts. Nice work done here.

Thank you so much 😊

**Pull request merged**